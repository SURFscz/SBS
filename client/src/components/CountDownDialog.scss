.count-down-content {
    span.counter {
        font-weight: 600;
    }
}