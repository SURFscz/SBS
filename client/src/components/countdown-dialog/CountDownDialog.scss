.count-down-dialog {
    @keyframes fade-out {
        0% {
            opacity: 0.95;
        }
        100% {
            opacity: 0.01;
        }
    }

    &.almost-there {
        animation: fade-out 5s ease-in;
    }

    .count-down-content {
        span.counter {
            font-weight: 600;
        }
    }
}
