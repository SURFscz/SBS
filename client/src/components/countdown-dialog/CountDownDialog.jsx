import React from "react";
import I18n from "../../locale/I18n";
import "./CountDownDialog.scss";
import {Modal} from "@surfnet/sds";
import DOMPurify from "dompurify";

export default function CountDownDialog({
                                            serviceName,
                                            isOpen = false,
                                            counter
                                        }) {

    const content = () => {
        return (
            <section className={"count-down-content"}>
                <p>
                    {I18n.t("countDownDialog.subTitle", {name: serviceName})}
                </p>
                <p dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("countDownDialog.info", {counter: counter}))}}/>
            </section>)
    }
    if (!isOpen) {
        return null;
    }

    return (
        <Modal
            confirm={null}
            subTitle={null}
            children={content()}
            title={I18n.t("countDownDialog.title", {name: serviceName})}
            className={`count-down-dialog ${counter < 5 ? "almost-there" : ""}`}
        />
    );

}
