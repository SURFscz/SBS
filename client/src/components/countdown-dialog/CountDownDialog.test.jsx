import React from "react";
import {render} from "@testing-library/react";
import CountDownDialog from "./CountDownDialog";

jest.mock("../../locale/I18n", () => ({
    t: (key, opts) => (opts && opts.seconds ? `${key} ${opts.seconds}` : key)
}));
jest.mock("@surfnet/sds", () => ({
    Modal: props => (
        <div data-testid="modal">
            <div>{props.title}</div>
            <div>{props.subTitle}</div>
            <div>{props.children}</div>
            <button disabled={props.confirmDisabled}>{props.confirmationButtonLabel}</button>
            <button>{props.cancelButtonLabel}</button>
        </div>
    )
}));

describe("CountDownDialog", () => {
    it("matches snapshot", () => {
        const {asFragment} = render(
            <CountDownDialog
                isOpen={true}
                close={jest.fn()}
                confirm={jest.fn()}
                title="Count down"
                subTitle="Counting down"
                confirmationButtonLabel="OK"
                cancelButtonLabel="Cancel"
                confirmDisabled={false}
                seconds={10}
            >
                <div>Dialog content</div>
            </CountDownDialog>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
