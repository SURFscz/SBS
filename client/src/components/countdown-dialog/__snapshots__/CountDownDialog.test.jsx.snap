// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDownDialog matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="modal"
  >
    <div>
      countDownDialog.title
    </div>
    <div />
    <div>
      <section
        class="count-down-content"
      >
        <p>
          countDownDialog.subTitle
        </p>
        <p>
          countDownDialog.info
        </p>
      </section>
    </div>
    <button />
    <button />
  </div>
</DocumentFragment>
`;
